subsys/net/lib/sockets/sockets_select.c

```diff
@@ -106,26 +73,59 @@ int zsock_select(int nfds, zsock_fd_set *readfds, zsock_fd_set *writefds,
 	int i, res, poll_timeout;
 	int num_pfds = 0;
 	int num_selects = 0;
+	int fd_no;
 
-	res = fd_set_to_pollfd(ZSOCK_POLLIN, readfds, pfds, &num_pfds);
-	if (res == -1) {
-		return -1;
-	}
+	fd_no = 0;
+	for (i = 0; i < STRUCT_MEMBER_ARRAY_SIZE(zsock_fd_set, bitset); i++) {
+		uint32_t bit_mask = 1;
+		uint32_t read_mask = 0, write_mask = 0, except_mask = 0;
+		uint32_t ored_mask;
 
-	res = fd_set_to_pollfd(ZSOCK_POLLOUT, writefds, pfds, &num_pfds);
-	if (res == -1) {
-		return -1;
-	}
+		if (readfds != NULL) {
+			read_mask = readfds->bitset[i];
+		}
+		if (writefds != NULL) {
```

Comment:
empty line before if, apply that below as well

Reply:
> empty line before if, apply that below as well

I replied previously already that here it's kind parallel statement, the same thing applied to 3 vars in row. Initially it was:

~~~
read_mask = readfds->bitset[i];
write_mask = writefds->bitset[i];
except_mask = exceptfds->bitset[i];
~~~

But turned out structure pointers need to be tested for NULL. So, added if's, the structure without the empty lines is still there, to emphasize they're effectively the same blocking of code doing similar thing.

If we have formal rule that after each "if" there should be empty line, the we should enforce it via checkpatch/etc. And we don't enforce that, then apparently just for the cases like this.

Reply:
This was actually addressed in the end.